### PR TITLE
:sparkles: Added jaeger to images so it runs out of the box

### PR DESCRIPTION
**Pull request by Parthiba-Hazra**
Fixes: #236

### CODE REVIEW

**Comment by pranavgaikwad** [comment]
@Parthiba-Hazra Thank you for your PR! Can we not build Jaeger and directly include the [jaeger/all-in-one](https://hub.docker.com/r/jaegertracing/all-in-one) image? Also, can we remove duplication in demo.Dockerfile and use the binary directly from base layer?

**Comment by Parthiba-Hazra** [comment]
@pranavgaikwad please review the changes.. sorry for the trouble actually I had to use command `all-in-one-linux` instead of `jaeger-all-in-one`. :)

File: demo.Dockerfile [hunk 1]
```diff
@@ -1,7 +1,15 @@
+FROM jaegertracing/all-in-one:latest AS jaeger-builder
```
Comment (pranavgaikwad) [inline demo.Dockerfile:1]: I don't think we need this line here, the binary should already be available in the image `quay.io/konveyor/analyzer-lsp`

File: demo.Dockerfile [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
+
 FROM quay.io/konveyor/analyzer-lsp
 
+COPY --from=jaeger-builder /go/bin/all-in-one-linux /usr/bin/
```
Comment (pranavgaikwad) [inline demo.Dockerfile:5]: Same as above, not needed

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]

File: demo.Dockerfile [hunk 3]
```diff
@@ -1,7 +1,15 @@
...
+
 WORKDIR /analyzer-lsp
 
 COPY rule-example.yaml /analyzer-lsp/rule-example.yaml
 COPY examples /analyzer-lsp/examples
 COPY open-source-libs.txt /analyzer-lsp/open-source-libs.txt
+
+EXPOSE 5775/udp 6831/udp 6832/udp 5778 16686 14268 9411
+
+ENTRYPOINT ["sh", "-c", "all-in-one-linux & sleep 5 && konveyor-analyzer --enable-jaeger=true --jaeger-endpoint=http://localhost:14268/api/traces && curl -o traces.json http://localhost:16686/api/traces?service=analyzer-lsp"]
```
Comment (pranavgaikwad) [inline demo.Dockerfile:15]: this `--jaeger-endpoint` is already the default, no need to explicitely pass https://github.com/konveyor/analyzer-lsp/blob/7cb1d667a4088a4ba11711491ff98987ce80c3ab/cmd/analyzer/main.go#L58

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]
Can we also enable jaeger by default? instead of having users to pass it, now that it's already in the image, it should just work

File: demo.Dockerfile (hunk 3 above)
Comment (Parthiba-Hazra) [reply demo.Dockerfile:15]: --enable-jaeger=true already there.. I think I just need to remove the --jaeger-endpoint flag?

Comment (pranavgaikwad) [reply demo.Dockerfile:15]: @Parthiba-Hazra I meant in the analysis CLI here https://github.com/konveyor/analyzer-lsp/blob/7cb1d667a4088a4ba11711491ff98987ce80c3ab/cmd/analyzer/main.go#L57

Comment (Parthiba-Hazra) [reply demo.Dockerfile]: Done.. thanks for the guidance :)

File: Dockerfile [hunk 4]
```diff
@@ -28,4 +32,6 @@ COPY provider_container_settings.json /analyzer-lsp/provider_settings.json
 
 WORKDIR /analyzer-lsp
 
-ENTRYPOINT ["konveyor-analyzer"]
+EXPOSE 5775/udp 6831/udp 6832/udp 5778 16686 14268 9411
```
Comment (pranavgaikwad) [inline Dockerfile:35]: @Parthiba-Hazra thank you for incorporating the changes. One last thing, I missed this earlier, sorry. Can we only expose the 16686 port, as all other ports are only used internally?

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 4 above)
Comment (Parthiba-Hazra) [reply Dockerfile]: done :+1:

**Review by pranavgaikwad — APPROVED** [review]

**Comment by pranavgaikwad** [comment]
@Parthiba-Hazra thank you! appreciate your patience on this. Going to to give this a quick test before merging.

**Comment by pranavgaikwad** [comment]
@shawn-hurley any objection on using jaeger image from docker.io?

**Comment by shawn-hurley** [comment]
None from me, I am a plus one

**Pull request merged**